### ISSUE TITLE
notcurses 2.3.15 build crashes on musl.

### ISSUE THREAD

**Issue by mobinmob** [dankamongmen/notcurses#2073]
distribution: void linux
COLORTERM="truecolor"
LANG="el_GR.UTF-8"
LANGUAGE="el_GR:el"
TERM="xterm-256color"
version :2.3.15
terminal: xfce4-terminal-0.8.10

The build process stops with the following error:
```
In file included from /builddir/notcurses-2.3.15/include/notcurses/notcurses.h:11,
                 from /builddir/notcurses-2.3.15/include/notcurses/direct.h:4,
                 from /builddir/notcurses-2.3.15/include/ncpp/Direct.hh:5,
                 from /builddir/notcurses-2.3.15/src/ls/main.cpp:16:
/usr/include/string.h:69:7: error: declaration of 'char* strndup(const char*, size_t)' has a different exception specifier
   69 | char *strndup (const char *, size_t);
      |       ^~~~~~~
In file included from /builddir/notcurses-2.3.15/src/ls/main.cpp:15:
/builddir/notcurses-2.3.15/src/compat/compat.h:82:7: note: from previous declaration 'char* strndup(const char*, size_t) noexcept'
   82 | char* strndup(const char* str, size_t size) NOEXCEPT;
      |       ^~~~~~~
[264/392] Building CXX object CMakeFiles/ncplayer.dir/src/player/play.cpp.o
ninja: build stopped: subcommand failed.
```
Full build logs can be found here:
https://github.com/void-linux/void-packages/pull/32563/checks?check_run_id=3363531667

**Comment by dankamongmen** [comment]
i know exactly how to fix this. i've also got some unit test breakage on fedora and alpine, so i'll likely cut a 2.3.16 pretty soon with the fix, though i'm also crushed a bit at work. latest should be saturday. i f you want to pick up the patch and apply it locally, i'll have that available within a few minutes, but it's your call -- most of the 2.3.15 changes were for microsoft windows; it's not the most important linux release ever.

**Comment by dankamongmen** [comment]
i'm surprised we didn't see this same failure on alpine, actually. what version of musl are you using in void?

**Comment by dankamongmen** [comment]
this ought be resolved by 6296a2480. thanks for the report!

**Comment by mobinmob** [comment]
> i'm surprised we didn't see this same failure on alpine, actually. what version of musl are you using in void?

We are still on 1.1.24 with some backported patches.

**Comment by mobinmob** [comment]
> this ought be resolved by [6296a24](https://github.com/dankamongmen/notcurses/commit/6296a2480f047b3d4517a8d4db9a9a43d523fe84). thanks for the report!

Thanks for the fix!

**Comment by dankamongmen** [comment]
no problem, as always appreciate the rapid and faithful feedback of my friends in void linux!
